src/Domain/User.php

```diff
@@ -10,94 +10,88 @@
 /**
  * Domain model for User within the Baikal system
  */
-final class User
-{
+final class User extends DomainObject {
+
     /**
-     * @var Username
+     * Unique ID
+     *
+     * @var mixed
      */
-    private $username;
+    public $id;
```

Comment:
For what it's worth, I carry the very unpopular opinion (within the current PHP metagame) that less code is better and actually results in more maintainability, instead of all the layers of complexity people keep on coming up with. 1 class per field is a big no-no for me. I can live with a bit of duplication if it means that more related code sits together in the same class. I'll add an interface or class when I need it, not if I anticipate needing it later.... if you catch my drift.

I hope that's not too frustrating, I do fully realize that I'm going against common sensibilities.